plugins/action_modify_record/src/databaseMethods.ts

```diff
@@ -1,56 +1,121 @@
+import { DBConnectType } from '../../../src/components/databaseConnect'
+
 const databaseMethods = (DBConnect: any) => ({
-  doesRecordExist: async (tableName: string, matchField: string, value: any) => {
+  getRecordId: async (tableName: string, matchField: string, value: any) => {
     const text = `
-      SELECT COUNT(*) FROM "${tableName}"
+      SELECT id FROM "${tableName}"
       WHERE "${matchField}" = $1
     `
     try {
       const result = await DBConnect.query({ text, values: [value] })
-      return result.rows[0].count > 0
+      return result?.rows[0]?.id || 0
     } catch (err) {
       console.log(err.message)
       throw err
     }
   },
-  updateRecord: async (
-    tableName: string,
-    matchField: string,
-    matchValue: any,
-    record: { [key: string]: any }
-  ) => {
+  updateRecord: async (tableName: string, id: string, record: { [key: string]: any }) => {
     const placeholders = DBConnect.getValuesPlaceholders(record)
     const matchValuePlaceholder = `$${placeholders.length + 1}`
     const text = `
       UPDATE "${tableName}" SET (${Object.keys(record)})
       = (${DBConnect.getValuesPlaceholders(record)})
```

Comment:
```suggestion
    const text = `
      UPDATE "${tableName}" SET (${Object.keys(record)})
      = (${placeholders})
```

Can probably use this variable here.

Although I'm trying to understand what this new WHERE statement is checking, what is in `matchValuePlaceholder`, and how an `id` can be compared with `$${placeholders.length + 1}` ?

Reply:
Hmmm, I think I understood now. this will be getting the last value from `values: [...Object.values(record), id]` which is the id, but not sure if this is in the right format, is it? Just trying to understand what is this one doing.

Reply:
f667700abb4426796a8d4664bb03cc6cdfa2e50d

> Hmmm, I think I understood now. this will be getting the last value from values: [...Object.values(record), id] which is the id, but not sure if this is in the right format, is it? Just trying to understand what is this one doing.

it's adding a new placeholder, see below `values: [...Object.values(record), id]`

Reply:
Yep. Got it : )